internal/observability/graph.go

```diff
@@ -309,9 +312,23 @@ type streamEdge struct {
 
 // newEdge constructs a new edge. Since Registries enforce metric uniqueness,
 // edges should be cached and re-used for each RPC.
-func newEdge(logger *zap.Logger, meter *metrics.Scope, tagToIgnore *metricsTagIgnore, req *transport.Request, direction string, rpcType transport.Type) *edge {
+func newEdge(logger *zap.Logger, meter *metrics.Scope, tagToIgnore *metricsTagIgnore, metricTagsDecorators []metricstagdecorators.MetricsTagsDecorators, req *transport.Request, direction string, rpcType transport.Type) *edge {
 	tags := tagToIgnore.tags(req, direction, rpcType)
 
+	if len(metricTagsDecorators) > 0 {
+		decoratorProperties := metricstagdecorators.DecoratorProperties{}
+
+		for _, decorator := range metricTagsDecorators {
+			decoratorTags := decorator.ProvideTags(decoratorProperties)
```

Comment:
I don't think I understand this completely on what you are trying to do here. Let's sync offline for this.

Reply:
remove DecoratorProperties